app/models/work.rb

```diff
@@ -197,6 +197,11 @@ def new_recipients_allow_gifts
 
   before_save :clean_and_validate_title, :validate_published_at, :ensure_revised_at
 
+  # If the work is becoming revealed, bump the revised-at date (AO3-3866)
+  before_save do
+    set_revised_at(Time.zone.today) if will_save_change_to_in_unrevealed_collection?
```

Comment:
To be precise, should we use `Time.current`?

Reply:
Good call, done. Somehow I got it into my head that these fields were date instead of datetime, but I see that is not the case.